docs/src/main/tut/typeclasses/applicative.md

```diff
@@ -0,0 +1,30 @@
+---
+layout: docs
+title:  "Applicative"
+section: "typeclasses"
+source: "algebird-core/src/main/scala/com/twitter/algebird/Applicative.scala"
+scaladoc: "#com.twitter.algebird.Applicative"
+---
+
+# Applicative
+
+*(Note - don't use this, use [cats](https://github.com/typelevel/cats))*
+
+Simple implementation of an `Applicative` type-class. There are many choices for the canonical second operation (`join`, `sequence`, `joinWith`, `ap`),
+all equivalent. For a `Functor` modeling concurrent computations with failure, like `Future`, combining results with `join` can save a lot of time over combining with `flatMap`. (Given two operations, if the second fails before the first completes, one can fail the entire computation right then. With `flatMap`, one would have to wait for the first operation to complete before failing it.)
```

Comment:
```
Given two operations, if the second fails before the first completes, one can fail the entire computation right then. With `flatMap`, one would have to wait for the first operation to complete before failing it.
```

This isn't quite true - if you want your `ap`/`zip` to agree with `flatMap`, you have to wait on the first one to fail, even if the 2nd fails, so that you can surface the error from the first. If the first fails, then you can bail out of the 2nd one early, but not vice versa.